7_drivers_munge/src/GCM_driver_utils.R

```diff
@@ -242,26 +245,125 @@ download_gcm_data <- function(out_file_template, query_geom,
   query_simplegeom <- sf_pts_to_simplegeom(query_geom_WGS84)
 
   # Build query_url
-  query_url <- sprintf("https://cida.usgs.gov/thredds/dodsC/notaro_%s_1980_1999", gcm_name)
-
-  # construct and submit query
-  gcm_job <- geoknife(
-    stencil = query_simplegeom,
-    fabric = webdata(
-      url = query_url,
-      variables = query_vars,
-      times = query_dates
+  query_url <- sprintf("https://cida.usgs.gov/thredds/dodsC/notaro_%s_%s",
+                       gcm_name, gcm_projection_period)
+
+  # Retry the GDP query multiple times if it doesn't work.
+  retry({
+    # construct and submit query
+    gcm_job <- geoknife(
+      stencil = query_simplegeom,
+      fabric = webdata(
+        url = query_url,
+        variables = query_vars,
+        times = query_dates
+      )
+      # The default knife algorithm is appropriate
     )
-    #TODO: should we specify the `knife`??
-  )
-  wait(gcm_job)
+    wait(gcm_job)
+    gcm_job_status <- check(gcm_job)$statusType
+  },
+  # Check the value of the last thing in the expr above (denoted
+  # by the ".") to decide if you should retry or not
```

Comment:
Thanks for adding such good comments - really helps to make clear here what is being checked